test/integration/sa_oauthclient_test.go

```diff
@@ -171,29 +180,108 @@ func runOAuthFlow(t *testing.T, clusterAdminClientConfig *restclient.Config, pro
+	authentictedAuthorizeResponse1, err := directHTTPClient.Transport.RoundTrip(authenticatedAuthorizeHTTPRequest1)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if authentictedAuthorizeResponse1.StatusCode != http.StatusFound {
+		response, _ := httputil.DumpResponse(authentictedAuthorizeResponse1, true)
+		t.Fatalf("unexpected status :\n %v", string(response))
+	}
+
+	// second we get a webpage with a prompt.  Yeah, this next bit gets nasty
+	authenticatedAuthorizeHTTPRequest2, err := http.NewRequest("GET", clusterAdminClientConfig.Host+authentictedAuthorizeResponse1.Header.Get("Location"), nil)
+	authenticatedAuthorizeHTTPRequest2.Header.Add("X-CSRF-Token", "csrf-01")
+	authenticatedAuthorizeHTTPRequest2.Header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte("harold:any-pass")))
+	authentictedAuthorizeResponse2, err := directHTTPClient.Transport.RoundTrip(authenticatedAuthorizeHTTPRequest2)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if authentictedAuthorizeResponse2.StatusCode != http.StatusOK {
+		response, _ := httputil.DumpResponse(authentictedAuthorizeResponse2, true)
+		t.Fatalf("unexpected status :\n %v", string(response))
+	}
+	// have to parse the page to get the csrf value.  Yeah, that's nasty, I can't think of another way to do it without creating a new grant handler
+	body, err := ioutil.ReadAll(authentictedAuthorizeResponse2.Body)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if !expectGrantSuccess {
+		if !strings.Contains(string(body), "requested illegal scopes") {
+			t.Fatalf("missing expected message: %v", string(body))
+		}
+		return
+	}
+	csrfMatches := grantCSRFRegex.FindStringSubmatch(string(body))
+	if len(csrfMatches) != 2 {
+		response, _ := httputil.DumpResponse(authentictedAuthorizeResponse2, false)
+		t.Fatalf("unexpected body :\n %v\n%v", string(response), string(body))
+	}
+	thenMatches := grantThenRegex.FindStringSubmatch(string(body))
+	if len(thenMatches) != 2 {
+		response, _ := httputil.DumpResponse(authentictedAuthorizeResponse2, false)
+		t.Fatalf("unexpected body :\n %v\n%v", string(response), string(body))
+	}
+	t.Logf("CSRF is %v", csrfMatches)
+	t.Logf("then is %v", thenMatches)
+
+	// third we respond and approve the grant, then let the transport follow redirects and give us the code
+	postBody := strings.NewReader(url.Values(map[string][]string{
+		"then":         []string{thenMatches[1]},
+		"csrf":         []string{csrfMatches[1]},
+		"client_id":    []string{oauthClientConfig.ClientId},
+		"user_name":    []string{"harold"},
+		"scopes":       []string{oauthClientConfig.Scope},
+		"redirect_uri": []string{clusterAdminClientConfig.Host},
+		"approve":      []string{"true"},
+	}).Encode())
+	authenticatedAuthorizeHTTPRequest3, err := http.NewRequest("POST", clusterAdminClientConfig.Host+"/oauth/approve", postBody)
```

Comment:
Maybe use the `OpenShiftApprovePrefix` constant here, instead of hardcoding the `/oauth/approve` endpoint.